react-app/src/components/Common/newbutton.js

```diff
@@ -1,158 +1,186 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addClink } from '../../features/clink';
 import 'antd/dist/antd.css';
 import '../../index.css';
 import { PlusOutlined } from '@ant-design/icons';
-import { Modal, Tabs, Button, Form, Input, Select } from 'antd';
+import { message, Modal, Tabs, Button, Form, Input, Select } from 'antd';
 
 const { Option } = Select;
 const { TabPane } = Tabs;
 
-// const layout = {
-//   layout: 'vertical'
-// };
+const NewButton = () => {
 
-class NewButton extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { loading: false, visible: false, form: 'bookmark' };
-  }
+  const dispatch = useDispatch()
+  const [bookmarkForm] = Form.useForm()
+  const [clinkForm] = Form.useForm()
+  const key = "formFeedback"
+
+  const [visible, setVisible] = useState(false);
+  const [loading, setLoading] = useState(false);
```

Comment:
If it's a boolean, I would rename this as setIsLoading and isLoading, same for visible